service/src/java/org/apache/hive/service/cli/operation/Operation.java

```diff
@@ -88,8 +89,14 @@ protected Operation(HiveSession parentSession, OperationType opType) {
   }
 
   protected Operation(HiveSession parentSession,
-      Map<String, String> confOverlay, OperationType opType) {
+                      Map<String, String> confOverlay, OperationType opType) {
+    this(parentSession, confOverlay, opType, false);
+  }
+
+  protected Operation(HiveSession parentSession,
+      Map<String, String> confOverlay, OperationType opType, boolean embedded) {
```

Comment:
I don't see this or SqlOperation called with a `true` embedded parameter

this one way to create a multi statement query thing - I wonder if we could have that as a first class citizen - and implement the old one as a single-instruction operation. what do you think?

Reply:
The SqlOperation is called with embedded=true when it is instantiated from a running hplsql script.

```
new SQLOperation(parentSession, statement, confOverlay, runAsync, queryTimeout, hplSqlMode());
```

Could elaborate more on that idea of having it as a first class citizen?